### PR TITLE
[docs] Consolidate instructions on server features

### LINKED ISSUE #4313: Easier to find `.headers` and `.sub` files documentations
I'm looking into writing wptserve based tests and the [easily findable documentation](https://github.com/w3c/web-platform-tests/blob/master/server-side.md) on that subject doesn't seem to mention `.headers` and `.sub` files which enable simpler conversion of existing Blink layout tests into wptserve tests. 

Would be great if those were mentioned there, with links towards appropriate documentation of those options.

### PR DESCRIPTION
**Pull request by jugglinmike**
The file `server-side.md` duplicates much of the information available
in the `docs/` directory. Remove `server-side.md`, preserving its unique
content in the relevant files within the `docs/` directory.

---

In addition to addressing general maintainability concerns (e.g.
`server-side.md` was missing newer `sub` values), focusing content in `docs`
makes it discoverable through the website's search functionality. This should
resolve gh-4313.

### CODE REVIEW

File: docs/writing-tests/server-features.md [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
 
-Certain test scenarios require more than just static HTML
-generation. This is supported through the
-[wptserve](http://wptserve.readthedocs.io) server, and controlled by
-[file name flags](file-names). Several scenarios in particular are common:
+* Tests that require cross-domain access
 
+* Tests that depend on setting specific headers or status codes
+
+* Tests that need to inspect the browser sent request
```
Comment (sideshowbarker) [inline docs/writing-tests/server-features.md]:
```suggestion
* Tests that need to inspect the browser-sent request
```

File: docs/writing-tests/server-features.md [hunk 2]
```diff
@@ -90,14 +109,18 @@ the file e.g. `test.html.sub.headers`.
    server-pipes
 ```
 
-For full control over the request and response the server provides the
-ability to write `.asis` files; these are served as literal HTTP
-responses. It also provides the ability to write [Python
+For full control over the request and response the server provides the ability
```
Comment (sideshowbarker) [inline docs/writing-tests/server-features.md]:
```suggestion
For full control over the request and response, the server provides the ability
```

File: docs/writing-tests/server-features.md [hunk 3]
```diff
@@ -90,14 +109,18 @@ the file e.g. `test.html.sub.headers`.
...
+to write `.asis` files; these are served as literal HTTP responses. In other
+words, they are sent byte for byte to the server without adding a HTTP status
```
Comment (sideshowbarker) [inline docs/writing-tests/server-features.md]:
```suggestion
words, they are sent byte-for-byte to the server without adding an HTTP status
```

**Review by sideshowbarker — APPROVED** [review]

**Comment by jugglinmike** [comment]
Thanks, @sideshowbarker!

File: docs/writing-tests/server-features.md [hunk 4]
```diff
@@ -1,12 +1,31 @@
...
+
+* Tests that require state to be stored on the server
+
+* Tests that require precise timing of the response.
+
+To make writing such tests possible, we are using a number of
+server-side components designed to make it easy to manipulate the
+precise details of the response:
+
+* *wptserve*, a custom python HTTP server.
```
Comment (gsnedders) [inline docs/writing-tests/server-features.md]:
```suggestion
* *wptserve*, a custom Python HTTP server.
```

(and elsewhere)

Comment (jugglinmike) [reply docs/writing-tests/server-features.md]: Done!

**Pull request merged**